tests/queryset_pickle/tests.py

```diff
@@ -153,3 +153,23 @@ def test_unsupported_unpickle(self):
         msg = "Pickled queryset instance's Django version 1.0 does not match the current version %s." % get_version()
         with self.assertRaisesMessage(RuntimeWarning, msg):
             pickle.loads(pickle.dumps(qs))
+
+    def test_queryset_evaluation(self):
+        """
+        Pickling a QuerySet.query with an __in=inner_qs lookup,
+        shouldn't evaluate inner_qs.
+        """
+        for i in range(1, 5):
+            Group(name='Group {}'.format(i)).save()
+
+        Event(title='Event 1', group=Group.objects.all()[0]).save()
+
+        groups = Group.objects.all()
+        events = Event.objects.filter(group__in=groups)
```

Comment:
> But to my knowledge passing just query instead of queryset for __in lookup isn't documented feature.

I knew I wasn't crazy, it [used to be documented](https://github.com/django/django/commit/f747b61c20cb2ab9bcbca2b272f3fba80cf39ac6#diff-5e313dd49d68a0a867d830a7f442d2a6L1042)!

The code clearly assumes both a `Query` and `QuerySet` can be passed so adding a test here for the former case makes sense IMHO.

`Event.objects.filter(group__in=groups.values('pk').query)` should do.